src/platform/intel/ace/lib/clk.c

```diff
@@ -11,9 +11,26 @@
 #include <sof/common.h>
 #include <sof/lib/memory.h>
 #include <rtos/spinlock.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clk, LOG_LEVEL_DBG);
 
 static SHARED_DATA struct clock_info platform_clocks_info[NUM_CLOCKS];
 
+static int select_cpu_freq(int clock, int hz)
+{
+	const struct device *clock_ctrl = device_get_binding("clkctl");
```

Comment:
I think the get_binding was deprecated etc. so you'd probably want to use something like in ace/lib/dma.c:
DEVICE_DT_GET(DT_NODELABEL(foobar))

Reply:
Right. There's no info about it in the docs for `device_get_binding`, but it's there in zephyrs release notes.